packages/blocks/src/lib/maplibre/index.svelte

```diff
@@ -0,0 +1,142 @@
+<!--
+  @component
+
+  Creates a maplibre-gl map that will fill its parent.
+
+  Children will mount once the map is fully loaded.
+
+  ```svelte
+    <MapLibre>
+      <MapLibreMarker lngLat={{ lng: 0, lat: 0 }} />
+    </MapLibre>
+  ```
+-->
+
+<svelte:options immutable />
+
+<script lang='ts'>
+
+import 'maplibre-gl/dist/maplibre-gl.css';
+
+import { onMount, createEventDispatcher, setContext, onDestroy } from 'svelte';
+import { writable } from 'svelte/store';
+import { Map, NavigationControl } from 'maplibre-gl';
+import { style } from './style';
+import type { LngLat } from '$lib';
+
+/** The minimum camera pitch. */
+export let minPitch = 0;
+
+/** The maximum camera pitch. */
+export let maxPitch = 60;
+
+/** The initial map zoom. */
+export let zoom = 9;
+
+/**
+ * The initial map center.
+ * 
+ * @default { lng: -73.984421, lat: 40.7718116 }
+ * The Viam Robotics office.
+ */
+export let center: LngLat = { lng: -73.984_421, lat: 40.771_811_6 };
+
+type Events = {
+  /** Fired after the map has been created. */
+  create: Map;
+  /** Fired when the map camera moves. */
+  move: Map;
+  /** Fired when the map resizes. */
+  resize: Map;
+}
+
+const dispatch = createEventDispatcher<Events>();
+const mapStore = writable<Map | undefined>(undefined);
+const centerStore = writable<LngLat>(center);
+const sizeStore = writable({ width: 0, height: 0 });
+const zoomStore = writable(zoom);
+
+setContext('map', mapStore);
+setContext('center', centerStore);
+setContext('size', sizeStore);
+setContext('zoom', zoomStore);
```

Comment:
Follow up on other comment: I think we should create all these stores and call `setContext` in a `provideMapLibre` function that is exported alongside `useMapLibre`

Reply:
Updated!